src/test/java/org/folio/rest/impl/TransactionTest.java

```diff
@@ -0,0 +1,158 @@
+package org.folio.rest.impl;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.ALLOCATION;
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.TRANSFER;
+import static org.folio.rest.util.MockServer.addMockEntry;
+import static org.folio.rest.util.TestEntities.FUND;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.folio.rest.jaxrs.model.Transaction;
+import org.folio.rest.util.TestEntities;
+import org.junit.jupiter.api.Test;
+
+public class TransactionTest extends ApiTestBase {
+  private static final Logger logger = LoggerFactory.getLogger(TransactionTest.class);
+
+  private final String dollars = "USD";
+  private final String FISCAL_YEAR_ID = "684b5dc5-92f6-4db7-b996-b549d88f5e4e";
+  private final String HIST_ID = "fb7b70f1-b898-4924-a991-0e4b6312bb5f";
+  private final String CANHIST_ID = "68872d8a-bf16-420b-829f-206da38f6c10";
+  private final String LATHIST_ID = "e6d7e91a-4dbc-4a70-9b38-e000d2fbdc79";
+  private final String ASIAHIST_ID = "55f48dc6-efa7-4cfe-bc7c-4786efe493e3";
+
+  @Test
+  public void testTransferAndAllocationWithMatchingAllocatedIds() throws Exception {
+    logger.info("=== Test transfer and allocation toFund.allocatedFromIds matches fromFund.allocatedToIds) - Success ===");
+    addMockEntry(FUND.name(), new JsonObject(getMockData("mockdata/funds/HIST.json")));
+    addMockEntry(FUND.name(), new JsonObject(getMockData("mockdata/funds/CANHIST.json")));
+
+    // transfer
+    Transaction transaction = createTransaction(TRANSFER)
+      .withFromFundId(HIST_ID)
+      .withToFundId(CANHIST_ID);
+    verifyPostResponse(TestEntities.TRANSACTIONS_TRANSFER.getEndpoint(), JsonObject.mapFrom(transaction), APPLICATION_JSON, 201);
+
+    // allocation
+    transaction.setTransactionType(ALLOCATION);
+    verifyPostResponse(TestEntities.TRANSACTIONS_ALLOCATION.getEndpoint(), JsonObject.mapFrom(transaction), APPLICATION_JSON, 201);
+  }
+
+  @Test
+  public void testTransferAndAllocationWithEmptyAllocatedIds() throws Exception {
+    logger.info("=== Test transfer and allocation with empty toFund.allocatedFromIds and fromFund.allocatedToIds - Success ===");
+    addMockEntry(FUND.name(), new JsonObject(getMockData("mockdata/funds/CANHIST.json")));
+    addMockEntry(FUND.name(), new JsonObject(getMockData("mockdata/funds/LATHIST.json")));
+
+    // transfer
+    Transaction transaction = createTransaction(TRANSFER)
+      .withFromFundId(CANHIST_ID)
+      .withToFundId(LATHIST_ID);
```

Comment:
LATHIST fund does not have allocatedTo, that means Transfer  to this fund should not be allowed. And response code must be 422

Reply:
According to the [comments](https://issues.folio.org/browse/MODFIN-110?focusedCommentId=70531&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-70531), it should be allowed